### PR TITLE
Accepting c2x compilers

### PR DESCRIPTION
**Pull request by Bernardin-MS**
Issue: #7471

There is a corresponding PR on the native side

### CODE REVIEW

File: Extension/c_cpp_properties.schema.json [hunk 1]
```diff
@@ -36,10 +36,14 @@
                             "c99",
                             "c11",
                             "c17",
+                            "c20",
```
Comment (bobbrow) [inline Extension/c_cpp_properties.schema.json]: I'm not sure if c20 is going to be a thing.  There was ambiguity between c17 and c18 at one point as well and we settled on just keeping one of them.  I "think" we just need c23 in the schema.

**Comment by michelleangela** [comment]
The html file for the IntelliSense Configuration UI (run command `C/C++: Edit Configurations (UI)` to see the UI)  will need to updating too.

https://github.com/microsoft/vscode-cpptools/blob/main/Extension/ui/settings.html

image

File: Extension/c_cpp_properties.schema.json (hunk 1 above)
Comment (Colengms) [reply Extension/c_cpp_properties.schema.json]: There is no c20 or gnu20 defined for gcc https://gcc.gnu.org/onlinedocs/gcc/C-Dialect-Options.html#C-Dialect-Options  and presumably clang as well, as it copies it's std's from gcc https://clang.llvm.org/docs/UsersManual.html  ("The supported modes for C are c89, gnu89, c94, c99, gnu99, c11, gnu11, c17, gnu17, c2x, gnu2x, and various aliases for those modes.")

Comment (michelleangela) [reply Extension/c_cpp_properties.schema.json]: Where did `c20` come from? @Bernardin-MS Is there a source that refers to `c20`?

Comment (Bernardin-MS) [reply Extension/c_cpp_properties.schema.json]: @michelleangela it was a placeholder based off a reference PR

@Colengms so c2x is the only argument we track, right?

Comment (Colengms) [reply Extension/c_cpp_properties.schema.json]: @Bernardin-MS I'm not sure I understand your question.  "c23" and "gnu23" are valid.  But, "c20" and "gnu20" aren't valid, so could you remove those?  :)

File: Extension/ui/settings.html [hunk 2]
```diff
@@ -581,10 +581,12 @@
         <div class="section-text" data-loc-id="c.standard.description">The version of the C language standard to use for IntelliSense. Note: GNU standards are only used to query the set compiler to get GNU defines, and IntelliSense will emulate the equivalent C standard version.</div>
         <div>
             <select name="inputValue" id="cStandard" class="select-default">
+                <option value="c2x">c2x</option>
```
Comment (michelleangela) [inline Extension/ui/settings.html]: Should this be `c23`? Is there a reason this is `c2x` and not consistent with the other settings? Note: this file is the UI for the c_cpp_properties.json. The values in both the UI and .json for IntelliSense configuration should be the same.

Same comment for `gnu2x`

**Review by michelleangela — CHANGES_REQUESTED** [review]

File: Extension/ui/settings.html (hunk 2 above)
Comment (Bernardin-MS) [reply Extension/ui/settings.html]: thx, typo

**Review by michelleangela — APPROVED** [review]

**Review by sean-mcmanus — APPROVED** [review]

**Review by Colengms — APPROVED** [review]

**Review by sean-mcmanus — APPROVED** [review]

**Review by sean-mcmanus — CHANGES_REQUESTED** [review]
This shouldn't get checked in until the C++ side is ready.

**Comment by bobbrow** [comment]
We're holding this one for 1.15

**Review by Dkhack — APPROVED** [review]

**Review by sean-mcmanus — APPROVED** [review]

**Pull request merged**